scm/pipeline.go

```diff
@@ -0,0 +1,106 @@
+package scm
+
+type (
+	ObjectAttributes struct {
```

Comment:
many vendors do not use numeric id which means this abstraction may break when we try to add bitbucket or others